templates/common_macros.html.j2

```diff
@@ -859,3 +859,69 @@
     {% endif %}
   {% endfor %}
 {% endmacro %}
+
+{%- macro render_functional_chains(chains) -%}
+  {%- for chain in chains -%}
+    <h3>{{ linked_name_with_icon(chain) }}</h3>
+    {{ chain.context_diagram }}
+    <p>{{ description(chain) | safe }}</p>
+  {%- endfor -%}
+{%- endmacro -%}
+
+{%- macro get_cap_type(obj) -%}
+  {%- if obj.layer.xtype.endswith("PhysicalArchitecture") -%}
+    CapabilityRealization
+  {%- elif obj.layer.xtype.endswith("LogicalArchitecture") -%}
+    Capability
+  {%- else -%}
+    OperationalCapability
+  {%- endif -%}
+{%- endmacro %}
+
+{%- macro render_functional_chain(model, object, type_name, cap_type, label) -%}
```

Comment:
Why do we pass `model` as argument here? I don't see a reason for that.

If you don't feel comfortable accessing a global inside a macro, there's still `object._model` as well.